Code/GraphMol/MolDraw2D/DrawShape.cpp

```diff
@@ -0,0 +1,552 @@
+                              padding)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+// ****************************************************************************
+DrawShapeDashedWedge::DrawShapeDashedWedge(const std::vector<Point2D> points,
+                                           const DrawColour &col1,
+                                           const DrawColour &col2, int atom1,
+                                           int atom2, int bond)
+    : DrawShape(points, 1, false, col1, false, atom1, atom2, bond),
+      col2_(col2) {
+  PRECONDITION(points_.size() == 3, "dashed wedge wrong points");
+  at1Cds_ = points[0];
+  buildLines();
+}
+
+// ****************************************************************************
+void DrawShapeDashedWedge::buildLines() {
+  // assumes this is the starting configuration, where the 3 points define
+  // the enclosing triangle of the wedge.
+  Point2D point = points_[0];
+  Point2D end1 = points_[1];
+  Point2D end2 = points_[2];
+  points_.clear();
+  lineColours_.clear();
+  Point2D e1 = end1 - point;
+  Point2D e2 = end2 - point;
+
+  unsigned int nDashes = 6;
+  double sideLen = e1.length();
+  double dashSep = sideLen / nDashes;
+  // don't have the dashes too far apart or too close together.
+  if (dashSep > 20.0) {
+    nDashes = sideLen / 20;
+  } else if (dashSep < 5.0) {
+    nDashes = sideLen / 5;
+    nDashes = nDashes < 3 ? 3 : nDashes;
+  }
+  for (unsigned int i = 1; i < nDashes + 1; ++i) {
+    Point2D e11 = point + e1 * (rdcast<double>(i) / nDashes);
+    Point2D e22 = point + e2 * (rdcast<double>(i) / nDashes);
+    points_.push_back(e11);
+    points_.push_back(e22);
+    if (i > nDashes / 2) {
+      lineColours_.push_back(col2_);
+    } else {
+      lineColours_.push_back(lineColour_);
+    }
+  }
+}
+
+// ****************************************************************************
+void DrawShapeDashedWedge::myDraw(MolDraw2D &drawer) const {
+  drawer.setFillPolys(false);
+  drawer.setActiveAtmIdx(atom1_, atom2_);
+  drawer.setActiveBndIdx(bond_);
+  for (size_t i = 0, j = 0; i < points_.size(); i += 2, ++j) {
+    if (drawer.drawOptions().splitBonds) {
+      if (i < points_.size() / 2) {
+        drawer.setActiveAtmIdx(atom1_);
+      } else {
+        drawer.setActiveAtmIdx(atom2_);
+      }
+    }
+    drawer.drawLine(points_[i], points_[i + 1], lineColours_[j],
+                    lineColours_[j], true);
+  }
+}
+
+// ****************************************************************************
+void DrawShapeDashedWedge::scale(const Point2D &scale_factor) {
+  DrawShape::scale(scale_factor);
+  at1Cds_.x *= scale_factor.x;
+  at1Cds_.y *= scale_factor.y;
+
+  // We may want to adjust the number of lines, so set everything up like
+  // the original triangle was passed in and rebuild.
+  Point2D point{at1Cds_};
+  Point2D end1{points_[points_.size() - 1]};
+  Point2D end2{points_[points_.size() - 2]};
+  points_ = std::vector<Point2D>{point, end1, end2};
```

Comment:
This could involve less copying:
```suggestion
  points_ = {at1Cds_, points_[points_.size() - 1], points_[points_.size() - 2]};
```